src/app/state/mod.rs

```diff
@@ -82,24 +79,28 @@ pub fn initialize<'app>(config_path: PathBuf) -> AppState<'app> {
         debug!("No config, using default");
         AppConfig::default()
     };
+    AppState::new(config)
+}
 
-    let tabs = Tabs::default();
-
-    let sql_tab_state = SQLTabState::new();
-    #[cfg(feature = "flightsql")]
-    let flightsql_tab_state = FlightSQLTabState::new();
-    let logs_tab_state = LogsTabState::default();
-    let history_tab_state = HistoryTabState::default();
+impl<'app> AppState<'app> {
```

Comment:
I just refactored the second half of this function into `AppState::new()` so I could pass in a `AppConfig` rather than having to create one from a directory